test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1263 @@
+
+      :arg parSafe: If `true`, this vector will use parallel safe operations.
+      :type parSafe: `param bool`
+    */
+    proc init(other: vector(?t), param parSafe=false) {
+      _checkType(t);
+      if !isCopyableType(this.type.eltType) then
+        compilerError("Cannot copy vector with element type that " +
+                      "cannot be copied");
+      this.eltType = t;
+      this.parSafe = parSafe;
+      this.complete();
+      _requestCapacity(other.size);
+      _commonInitFromIterable(other);
+    }
+
+    /*
+      Initializes a vector containing elements that are copy initialized from
+      the elements contained in another vector.
+
+      :arg other: The vector to initialize from.
+    */
+    proc init=(other: vector(this.type.eltType)) {
+      _checkType(this.type.eltType);
+      if !isCopyableType(this.type.eltType) then
+        compilerError("Cannot copy vector with element type that " +
+                      "cannot be copied");
+      this.eltType = this.type.eltType;
+      this.parSafe = this.type.parSafe;
+      this.complete();
+
+      _requestCapacity(other.size);
+      _commonInitFromIterable(other);
+    }
+
+    /*
+      Initializes a vector containing elements that are copy initialized from
+      the elements contained in an array.
+
+      Used in new expressions.
+
+      :arg other: The array to initialize from.
+
+      :arg parSafe: If `true`, this vector will use parallel safe operations.
+      :type parSafe: `param bool`
+    */
+    proc init(other: [?d] ?t, param parSafe=false) {
+      _checkType(t);
+      if !isCopyableType(t) then
+        compilerError("Cannot construct vector from array with element " +
+                      "type that cannot be copied");
+
+      this.eltType = t;
+      this.parSafe = parSafe;
+      this.complete();
+      _requestCapacity(other.size);
+      _commonInitFromIterable(other);
+    }
+
+    /*
+      Initializes a vector containing elements that are copy initialized from
+      the elements yielded by a range.
+
+      Used in new expressions.
+
+      .. note::
+
+        Attempting to initialize a vector from an unbounded range will trigger
+        a compiler error.
+
+      :arg other: The range to initialize from.
+
+      :arg parSafe: If `true`, this list will use parallel safe operations.
+      :type parSafe: `param bool`
+    */
+    proc init(other: range(?t), param parSafe=false) {
+      _checkType(t);
+      this.eltType = t;
+      this.parSafe = parSafe;
+
+      if !isBoundedRange(other) {
+        param e = this.type:string;
+        param f = other.type:string;
+        param msg = "Cannot init " + e + " from unbounded " + f;
+        compilerError(msg);
+      }
+
+      this.complete();
+      _requestCapacity(other.size);
+      _commonInitFromIterable(other);
+    }
+    /*
+      Initializes a vector containing elements that are copy initialized from
+      the elements contained in another list.
```

Comment:
> Though, should there be a proc List.init=(other: vector(...)) and proc List.init(other: vector(...))?

I think it could be done when the new List interface is introduced.

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/61f46cc650cbf0285f58211c0e44e50f9c50dac4

Reply:
@cassella -- resolved?